rdflib/tools/defined_namespace_creator.py

```diff
@@ -0,0 +1,182 @@
+"""
+This rdflib Python script creates a DefinedNamespace Python file from a given RDF file
+
+It is a very simple script: it finds all things defined in the RDF file within a given
+namespace:
+
+    <thing> a ?x
+
+    where ?x is anything and <thing> starts with the given namespace
+
+Nicholas J. Car, Dec, 2021
+"""
+import sys
+from pathlib import Path
+import argparse
+import datetime
+
+sys.path.append(str(Path(__file__).parent.absolute().parent.parent))
+
+from rdflib import Graph, util
+from rdflib.namespace import DCTERMS, OWL, RDFS, SKOS
+
+
+def get_input_format(file_path):
+    input_format = util.guess_format(str(file_path))
+    if input_format is None:
+        str_path = str(file_path)
+        if str_path.endswith("json-ld") or str_path.endswith("jsonld"):
+            input_format = "json-ld"
+        else:
+            raise Exception(
+                "ERROR: Cannot guess the RDF format of input file {}".format(
+                    file_path)
+            )
+
+    return input_format
+
+
+def validate_namespace(namespace):
+    if not namespace.endswith(("/", "#")):
+        raise ValueError("The supplied namespace must end with '/' or '#'")
+
+
+def validate_object_id(object_id):
+    for c in object_id:
+        if not c.isupper():
+            raise ValueError("The supplied object_id must be an all-capitals string")
+
+
+def get_classes(g, target_namespace):
```

Comment:
Is this being used? Can't see where, but I may be overlooking it.

Reply:
Actually no! I started thinking about getting Classes & Properties - the main things in ontologies - but then reverted to a simpler collection of things in the namespace. I've left this function there because I hate deleting code but I will mark it unused.